component/prometheus/kubernetes/podmonitors/configGenerator.go

```diff
@@ -0,0 +1,176 @@
+package podmonitors
+
+// SEE https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/prometheus/promcfg.go
+
+import (
+	"fmt"
+	"regexp"
+
+	v1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	commonConfig "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	promk8s "github.com/prometheus/prometheus/discovery/kubernetes"
+	"github.com/prometheus/prometheus/model/relabel"
+)
+
+type configGenerator struct {
```

Comment:
nit: I'm not sure why some methods are exported and some aren't. For unexported types like `configGenerator`, I prefer to use exported methods to indicate "the caller is expected to use these" and unexported methods to indicate "the caller is not expected to use these."